core/runtime/src/main/java/org/apache/isis/core/runtime/system/persistence/adaptermanager/ObjectAdapterContext.java

```diff
@@ -0,0 +1,470 @@
+    
+    public static ObjectAdapterContext openContext(
+            ServicesInjector servicesInjector, 
+            AuthenticationSession authenticationSession, 
+            SpecificationLoader specificationLoader, 
+            PersistenceSession persistenceSession) {
+        final ObjectAdapterContext objectAdapterContext = 
+                new ObjectAdapterContext(servicesInjector, authenticationSession, 
+                        specificationLoader, persistenceSession);
+        objectAdapterContext.open();
+        return objectAdapterContext;
+    }
+    
+    private final PojoAdapterHashMap pojoAdapterMap = new PojoAdapterHashMap();
+    private final OidAdapterHashMap oidAdapterMap = new OidAdapterHashMap();
+    private final PersistenceSession persistenceSession; 
+    private final ServicesInjector servicesInjector;
+    private final SpecificationLoader specificationLoader;
+    private final ObjectAdapterContext_Consistency consistencyMixin;
+    private final ObjectAdapterContext_ObjectAdapterProvider objectAdapterProviderMixin;
+    private final ObjectAdapterContext_AdapterManager adapterManagerMixin;
+    private final ObjectAdapterContext_MementoSupport mementoSupportMixin;
+    
+    private ObjectAdapterContext(
+            ServicesInjector servicesInjector, 
+            AuthenticationSession authenticationSession, 
+            SpecificationLoader specificationLoader, 
+            PersistenceSession persistenceSession) {
+        
+        this.consistencyMixin = new ObjectAdapterContext_Consistency(this);
+        this.objectAdapterProviderMixin = new ObjectAdapterContext_ObjectAdapterProvider(this, persistenceSession);
+        this.adapterManagerMixin = new ObjectAdapterContext_AdapterManager(this, persistenceSession);
+        this.mementoSupportMixin = new ObjectAdapterContext_MementoSupport(this, persistenceSession);
+        
+        this.persistenceSession = persistenceSession;
+        this.servicesInjector = servicesInjector;
+        this.specificationLoader = specificationLoader;
+        
+        this.objectAdapterFactories = new ObjectAdapterContext_Factories(
+                authenticationSession, 
+                specificationLoader, 
+                persistenceSession);
+    }
+
+    // -- LIFE-CYCLING
+    
+    public void open() {
+        oidAdapterMap.open();
+        pojoAdapterMap.open();
+        initServices();
+    }
+    
+    public void close() {
+        
+        try {
+            oidAdapterMap.close();
+        } catch(final Throwable ex) {
+            // ignore
+            ObjectAdapterLegacy.LOG.error("close: oidAdapterMap#close() failed; continuing to avoid memory leakage");
+        }
+
+        try {
+            pojoAdapterMap.close();
+        } catch(final Throwable ex) {
+            // ignore
+            ObjectAdapterLegacy.LOG.error("close: pojoAdapterMap#close() failed; continuing to avoid memory leakage");
+        }
+    }
+    
+    // -- CACHING
```

Comment:
I see we do still have our caches, for the moment at least.

Reply:
yes, the point of merging this current state is to reduce the scope of java classes that are involved with coming changes with ISIS-1976